File: frontend/prest/src/app/models/user/UserUpdatePayload.ts
import {Payload} from "../common/Payload";
import {Commentary} from "@commentary/Commentary";
import {UserRestaurant} from "@userRestaurant/UserRestaurant";
import {Address} from "@address/Address";

export interface UserUpdatePayload extends Payload
{
  id_user: number;
  nickname: string;
  addresses: Address[];
}

File: Prest/src/main/java/be/prest/Payload/Update/UserUpdatePayload.java
package be.prest.Payload.Update;

import be.prest.Authentification.entity.*;
import be.prest.Entities.Address;
import be.prest.Entities.Commentary;
import be.prest.Entities.UserRestaurant;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class UserUpdatePayload {
    private int id_user;
    private String nickname;
    private List<Commentary> commentaries;
    private List<UserRestaurant> userRestaurants;
    private List<Address> addresses;
    private Credential credential;

}
